prog/encoding.go

```diff
@@ -593,12 +597,18 @@ func (p *parser) parseArgString(t Type, dir Dir) (Arg, error) {
 		p.eatExcessive(true, "wrong string arg")
 		return t.DefaultArg(dir), nil
 	}
-	data, err := p.deserializeData()
+	data, b64, err := p.deserializeData()
 	if err != nil {
 		return nil, err
 	}
+	if !typ.IsCompressed() && b64 {
+		data, err = Decompress(data)
```

Comment:
Not sure if it's much of a problem speed-wise, but here you decompress the data so that it would immediately be compressed in `MakeDataArg` at the bottom of the function.

Reply:
Only non-compressed types would be decompressed, so when stored these should not be compressed again I believe. But with the move to decouple Base64 and compression, this should no longer be an issue.